feat(network): automatic Antenna Criteria check

LeonVreling opened the pull request.

LeonVreling committed:
feat(network): Add a first draft for automatic AC check of events

LeonVreling committed:
chore(network): Add placeholder for all Antenna Criteria

LeonVreling committed:
feat(network): Add toggle to display detailed information

LeonVreling committed:
feat(network): Add toggle to display only endangered Locals

LeonVreling committed:
chore(network): Fix lint

LeonVreling committed:
chore(network): Restructure code to use methods and promises

LeonVreling committed:
feat(network): Add automatic check for the board election

LeonVreling committed:
fix(network): Display last event date instead of deadline for new event

LeonVreling committed:
feat(network): Add basic logic for different Local types

LeonVreling committed:
feat(network): Add automatic check for members list submission

LeonVreling committed:
fix(network): Add more placeholder information

LeonVreling committed:
feat(network): Add option to select Agora (instead of most recent)

LeonVreling committed:
feat(network): Check if event is within 2 years of selected Agora

LeonVreling committed:
chore(network): Add auto select to most recent Agora

LeonVreling committed:
feat(network) Check if board election is within 1 year of the Agora

LeonVreling committed:
chore(boards): Update name endpoint for backend consistency

LeonVreling committed:
feat(network): Add base modal to edit Antenna Criteria

LeonVreling committed:
feat(network): Add option to set communication AC fulfilment

LeonVreling committed:
feat(network): Display the correct data for AC fulfilment

LeonVreling committed:
feat(network): Add option to give exception

LeonVreling committed:
feat(network): Add modal to display comments

LeonVreling committed:
chore(network): Clean up code

LeonVreling committed:
feat(network): Add option to set Agora attendance AC fulfilment

LeonVreling committed:
feat(network): Add option to set DP and FR AC fulfilment

Also updated the info modal a bit for the comments

LeonVreling committed:
feat(network) Add option to set membership fee AC fulfilment

Also updated the logic, oops

LeonVreling committed:
chore(network): Set permissions for the different roles correctly

LeonVreling committed:
feat(network): Add ND override for automatic fields

LeonVreling committed:
fix(network): Update look of modal

LeonVreling committed:
chore(network): Lint

LeonVreling commented:
### List of used permissions:

**_NetCom:_**
- global:manage_network:antenna_criteria
- global:manage_network:communication

**_ND:_**
- _All of NetCom_
- global:manage_network:communication_exception
- global:manage_network:board_election
- global:manage_network:members_list
- global:manage_network:events
- global:manage_network:agora_attendance
- global:manage_network:development_plan
- global:manage_network:fulfilment_report

**_FD:_**
- global:manage_network:antenna_criteria
- global:manage_network:membership_fee


EDIT: Update according to commit [#ecac9ab](https://github.com/AEGEE/frontend/pull/2036/commits/ecac9ab562952013996eace0ff3f333eff6b3ba0)

LeonVreling committed:
Merge branch 'stable' into ac-check

LeonVreling committed:
fix(network): Fix copy paste error

NetCom should no longer be allowed to give exceptions to `communication`

LeonVreling commented:
@WikiRik I just saw there are still two TODO's in the code for the checking of the `Events` criterion, but I think it would still be good to have a review since the list of changes is quite long 😁

WikiRik commented:
Yes, I think a review is good to do. I should be able to work on that this weekend

LeonVreling committed:
fix(network): Make permission naming consistent

WikiRik commented:
@LeonVreling can you make a PR in core for the new permissions (and the other permissions added in network earlier) and NetCom/ND/FD users? https://github.com/AEGEE/core/blob/stable/scripts/seed.js

WikiRik commented on src/router/routes.js:
```diff
@@ -841,6 +841,14 @@ module.exports = [
   },
+  {
+    name: 'oms.network.ac_check',
+    path: '/antenna_criteria_check',
```
```suggestion
    path: '/network/ac_check',
```

WikiRik commented on src/views/network/AntennaCriteriaCheck.vue:
```diff
@@ -0,0 +1,347 @@
+    fetchAgorae () {
+      this.isLoading = true
+      this.axios.get(this.services['statutory'], { params: { type: 'agora' } }).then((response) => {
+        this.agorae = response.data.data
```
Benefit of testing it on a relatively empty database is seeing how it works without any Agorae. Can you make it throw an error here if `this.agorae` is empty? Now it shows an error for 'Could not fetch bodies'

WikiRik reviewed (commented):
I'm going to take some time to review the backend PRs first and then set up more data to test with before I review this further

---

This is what I see if I only have the NetCom permissions created (and all that is in the core seeder), is that correct?
![image](https://github.com/user-attachments/assets/6f88efa6-359f-4ace-af14-45771024994d)

LeonVreling commented:
> @LeonVreling can you make a PR in core for the new permissions (and the other permissions added in network earlier) and NetCom/ND/FD users? https://github.com/AEGEE/core/blob/stable/scripts/seed.js

The pull request for core can be found at https://github.com/AEGEE/core/pull/972

LeonVreling committed:
chore(network): Throw error when no Agora exists

Also changes the url of the page

LeonVreling committed:
feat(network): Check for all events for the AC fulfilment

LeonVreling committed:
chore(network): Fix lint

LeonVreling committed:
fix(network): Only change AC you are allowed to change

LeonVreling committed:
chore(network): Fix lint

WikiRik commented on src/views/network/AntennaCriteriaModal.vue:
```diff
@@ -0,0 +1,251 @@
+      await Promise.all(promises).then(() => {
+        this.isLoading = false
+        this.showSuccess('Antenna Criteria fulfilment updated.')
+        // this.router.go(0)
```
```suggestion
```

WikiRik commented on src/views/core/bodies/ViewBoards.vue:
```diff
@@ -109,7 +128,7 @@ export default {
     fetchData () {
       // Get boards of the body
-      this.axios.get(this.services['network'] + '/bodies/' + this.$route.params.id + '?sort=start_date&direction=desc').then((response) => {
+      this.axios.get(this.services['network'] + '/bodies/' + this.$route.params.id + '/boards?sort=start_date&direction=desc').then((response) => {
```
I think I need to merge https://github.com/AEGEE/network/pull/599 before this works. Maybe move the changes to this file to another PR so we can merge the actual AC checks first

WikiRik replied:
Oh wait, it needs that PR either way to properly display the AC page. We can keep this

LeonVreling replied:
Ah I do agree that it would have been cleaner to make this a separate PR

LeonVreling committed:
chore(network): Add back reload after modal submission

I commented this out for easier local testing, but it should be part of the final release for consistency with the rest of the project

WikiRik committed:
Merge branch 'stable' into ac-check

WikiRik reviewed (approved):
We shouldn't deploy this until https://github.com/AEGEE/network/pull/599 is merged, but this looks good

serge1peshcoff commented:
:tada: This PR is included in version 1.37.0 :tada:

The release is available on [GitHub release](https://github.com/AEGEE/frontend/releases/tag/1.37.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
